src/visualization/ColorPalette/ColorMap.cc

```diff
@@ -0,0 +1,284 @@
+#include <string>
+#include <vector>
+#include <SFML/Graphics/Color.hpp>
+#include <array>
+#include <cmath>
+#include "ColorMap.h"
+#include <iostream>
+
+using namespace std;
+using namespace sf;
+
+/**
+ * This Method will compute in terms of the argument sequences and the k, all kmers we are looking for and call
+ * the method which compute us in terms of this number a Colormap. Then the method will compute
+ * all different Kmers and push them in a List. We'll use this list to initialize differentKmers.
+ */
+ColorMap::ColorMap(vector<string>& Sequences, int k){
+    ///< This is a variable which save the number of Kmers per Sequence which we have
+    vector<int> SequenceSizes;
+    for (int i=0;i<Sequences.size();i++)
+        SequenceSizes[i] = (int)floor((Sequences[i].size())/k);
+
+    ///< This is a variable which save the number of Kmers
+    int numbOfKmer = 0;
+    for (int i=0;i<SequenceSizes.size();i++)
+        numbOfKmer += SequenceSizes[i];
+    updateColorList(numbOfKmer);
+    ///< This variable is the output List
+    vector<string> kmerDeclarations;
+
+    ///< a boolean for the "got this kmer a color?"-Check
+    bool matched = false;
+    ///< a index for the k-mers per sequence
+    int index;
+    
+    /**
+     * This part of the method will push all Kmers in the List, which are not in the list already.
+     */
+    for (int i=0;i<Sequences.size();i++) {
+        index = 0;
+        while (index<SequenceSizes[i]) {
+            ///< Comparing all k-mers to know if the Kmer is already in the list
```

Comment:
Another data type is better than a vector that needs to be scanned front to back: `set` or `unordered_map`